pkg/segment/structs/evaluationstructs.go

```diff
@@ -1607,13 +1616,56 @@ func (self *NumericExpr) Evaluate(fieldToValue map[string]utils.CValueEnclosure)
 				return 0, fmt.Errorf("NumericExpr.Evaluate: cannot convert '%v' to number with base %d", strValue, base)
 			}
 			return float64(number), nil
+		case "relative_time":
+			if self.Left == nil {
+				return 0, fmt.Errorf("NumericExpr.Evaluate: relative_time operation requires a non-nil left operand")
+			}
+
+			var epochTime int64
+			var err error
+			// 1. Direct epoch input (e.g., relative_time(1234567890, "-1d")) is stored in self.Value.
+			// 2. Epoch from another command (e.g., relative_time(now(), "-1d")) after evaluation is stored in left, leaving self.Value as an empty string.
+			if self.Value != "" {
+				epochTime, err = strconv.ParseInt(self.Value, 10, 64)
+				if err != nil {
+					return 0, fmt.Errorf("NumericExpr.Evaluate: failed to parse epoch time from Value %v, err: %v", self.Value, err)
+				}
+			} else if left >= 0 {
+				epochTime = int64(left)
+			} else {
+				return 0, fmt.Errorf("NumericExpr.Evaluate: relative_time operation requires a valid timestamp")
+			}
+
+			relTime, err := CalculateAdjustedTimeForRelativeTimeCommand(self.RelativeTime, time.Unix(epochTime, 0))
+			if err != nil {
+				return 0, fmt.Errorf("NumericExpr.Evaluate: error calculating relative time: %v", err)
+			}
 
+			return float64(relTime), nil
 		default:
 			return 0, fmt.Errorf("NumericExpr.Evaluate: unexpected operation: %v", self.Op)
 		}
 	}
 }
 
+func CalculateAdjustedTimeForRelativeTimeCommand(timeModifier RelativeTimeExpr, currTime time.Time) (int64, error) {
+	var err error
```

Comment:
Maybe this function should be moved into a util package. What do you think?

Reply:
Yes Andrew, I moved it to `utils` package.